tests/functional_tests/ffront_tests/test_type_deduction.py

```diff
@@ -75,14 +75,18 @@ def type_info_cases() -> list[tuple[Optional[ct.SymbolType], dict]]:
     ]
 
 
-def accept_args_cases():
+def callable_type_info_cases():
     # reuse all the other test cases
     not_callable = [
-        (symbol_type, [], {}, [r"Expected a callable type, but got "])
+        (symbol_type, [], {}, [r"Expected a callable type, but got "], None)
         for symbol_type, attributes in type_info_cases()
         if not isinstance(symbol_type, ct.CallableType)
     ]
 
+    IDim = Dimension("I")
+    JDim = Dimension("J")
+    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
+
     bool_type = ct.ScalarType(kind=ct.ScalarKind.BOOL)
     float_type = ct.ScalarType(kind=ct.ScalarKind.FLOAT64)
     int_type = ct.ScalarType(kind=ct.ScalarKind.INT64)
```

Comment:
int32 the one mostly used in ICON and some times we had to change some things to account for that. Could you test this case as well?

Reply:
The actual type here doesn't matter, I could have equally used just `FLOAT64` and `FLOAT32`. The argument checking is and should never be dependent on what `ScalarType` is used in the signature. What matters is that different types. If in the implementation I screw up the order of the arguments, but my test would be for
```
def f(a: float, b: float):
    ...

f(some_float, some_float)
```
then the test would pass. Using different argument types however helps:
```
def f(a: int, b: float):
    ...

f(some_float, some_int)
```

Reply:
:+1: